src/Microsoft.PowerShell.Commands.Management/commands/management/Service.cs

```diff
@@ -1681,189 +1611,137 @@ protected override void ProcessRecord()
-                                password = Marshal.SecureStringToCoTaskMemUnicode(Credential.Password);
-                            }
-                            bool succeeded = NativeMethods.ChangeServiceConfigW(
-                                hService,
-                                NativeMethods.SERVICE_NO_CHANGE,
-                                dwStartType,
-                                NativeMethods.SERVICE_NO_CHANGE,
-                                null,
-                                null,
-                                IntPtr.Zero,
-                                null,
-                                username,
-                                password,
-                                DisplayName
-                                );
-                            if (!succeeded)
-                            {
-                                int lastError = Marshal.GetLastWin32Error();
-                                Win32Exception exception = new Win32Exception(lastError);
-                                WriteNonTerminatingError(
-                                    service,
-                                    exception,
-                                    "CouldNotSetService",
-                                    ServiceResources.CouldNotSetService,
-                                    ErrorCategory.PermissionDenied);
-                                continue;
-                            }
-                        } // modify startup type or display name
+                    bool status = NativeMethods.ChangeServiceConfig2W(
+                        hService,
+                        NativeMethods.SERVICE_CONFIG_DESCRIPTION,
+                        buffer);
 
-                        NativeMethods.SERVICE_DESCRIPTIONW sd = new NativeMethods.SERVICE_DESCRIPTIONW();
-                        sd.lpDescription = Description;
-                        int size = Marshal.SizeOf(sd);
-                        IntPtr buffer = Marshal.AllocCoTaskMem(size);
-                        Marshal.StructureToPtr(sd, buffer, false);
+                    if (!status)
+                    {
+                        int lastError = Marshal.GetLastWin32Error();
+                        Win32Exception exception = new Win32Exception(lastError);
+                        WriteNonTerminatingError(
+                            service,
+                            exception,
+                            "CouldNotSetServiceDescription",
+                            ServiceResources.CouldNotSetServiceDescription,
+                            ErrorCategory.PermissionDenied);
+                    }
 
-                        bool status = NativeMethods.ChangeServiceConfig2W(
-                            hService,
-                            NativeMethods.SERVICE_CONFIG_DESCRIPTION,
-                            buffer);
+                    // If '-Status' parameter is specified, do the necessary action to bring about the desired result
```

Comment:
I do not like this comment. It says if -Status, do something to get what you want. Can we be more specific or remove this comment.

Reply:
Sure, will change it to `// Handle '-Status' parameter`.